utils: upgrade string to uint32 function

Igor-Misic opened the pull request:
- add a new pow of 10 functions with the lookup table
- add guards for uint32 overflow
- add guards for non-digit characters
- add more tests

jciberlin commented on Inc/utils.h:
```diff
@@ -37,7 +37,8 @@
 
-uint32_t Utils_StringToUint32(const char* buf, const uint32_t lenght);
+bool Utils_QuickUint32Pow10(const uint8_t exponent, uint32_t* result);
+bool Utils_StringToUint32(const char* str, const uint8_t str_lenght, uint32_t* integer);
```
str_length

jciberlin commented on Src/utils.c:
```diff
@@ -34,20 +34,75 @@
+    bool check_overflow = false;
+    uint8_t i = 0U;
+    *integer = 0U;
+    uint8_t lenght = str_lenght;
```
length

jciberlin commented on Tests/test_utils.c:
```diff
@@ -20,6 +22,7 @@ TEST_TEAR_DOWN(Utils) {
 }
 
 TEST_GROUP_RUNNER(Utils) {
+	RUN_TEST_CASE(Utils, Utils_QuickUint32Pow10);
```
Use spaces instead of tab. We need to enable formatter for tests as well.

Igor-Misic replied:
> We need to enable formatter for tests as well.

Definitely

jciberlin commented on Tests/test_utils.c:
```diff
@@ -30,10 +33,72 @@ TEST_GROUP_RUNNER(Utils) {
 
+TEST(Utils, Utils_QuickUint32Pow10) {
+    uint32_t result = 0U;
+    bool success = false;
```
Not needed. We can use TEST_ASSERT_TRUE/FALSE instead of TEST_ASSERT. In my opinion, it looks better.

jciberlin commented on Tests/test_utils.c:
```diff
@@ -30,10 +33,72 @@ TEST_GROUP_RUNNER(Utils) {
-    TEST_ASSERT_EQUAL_UINT32(Utils_StringToUint32("0", 1U), 0U);
-    TEST_ASSERT_EQUAL_UINT32(Utils_StringToUint32("2165217", 7U), 2165217U);
+    uint32_t ui32_number = 0U;
+    bool success = false;
```
Not needed. Same as my previous comment for previous test case.

jciberlin reviewed (commented):
It looks good to me. Good job. I added some minor comments.

Igor-Misic committed:
utils: upgrade string to uint32 function

- add a new pow of 10 functions with the lookup table
- add guards for uint32 overflow
- add guards for non-digit characters
- add more tests
